src/internal.c

```diff
@@ -25916,8 +25995,6 @@ static int HashSkeData(WOLFSSL* ssl, enum wc_HashType hashType,
                                 ssl->buffers.sig.length,
                                 ssl->buffers.digest.buffer,
                                 ssl->buffers.digest.length);
-        XFREE(ssl->buffers.sig.buffer, ssl->heap, DYNAMIC_TYPE_SIGNATURE);
```

Comment:
It was not moved, it was removed because `ssl->buffers.sig.buffer` is needed later by the `ProcessServerSigKexCb` callback. The reason why its okay to remove is that `FreeKeyExchange()` will free it.

Reply:
It was not moved, it was removed because `ssl->buffers.sig.buffer` is needed later by the `ProcessServerSigKexCb` callback. The reason why its okay to remove is that `FreeKeyExchange()` will free it.

Reply:
I think it is important to free memory early if possible. Can you please still free it if the `ProcessServerSigKexCb` is not set?